inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -87,30 +87,17 @@ def get_all_single_schedules_in_priority_order(self) -> List[SingleSchedule]:
         @return: List of SingleSchedule object by priority in ascending order
         """
         try:
-            sql = ''' SELECT priority, schedule_id, job_id, status FROM single_schedule_job WHERE status IS NULL ORDER BY priority ASC; '''
+            sql = ''' SELECT ssj.priority, ssj.schedule_id, ssj.task_id, j.job_id FROM single_schedule_job ssj  JOIN job j ON ssj.task_id=j.task_id WHERE ssj.status IS NULL ORDER BY priority ASC; '''
             self._cursor.execute(sql)
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Optimize the database clearing method by using a transaction to wrap all delete operations.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        try:
            self._conn.execute('BEGIN')
            self._conn.execute('DELETE FROM single_schedule_job;')
            self._conn.execute('DELETE FROM repeated_schedule_job;')
            self._conn.execute('DELETE FROM single_schedule;')
            self._conn.execute('DELETE FROM repeated_schedule;')
            self._conn.execute('DELETE FROM job;')
            self._conn.execute('COMMIT')
        except sqlite3.Error as e:
            self._conn.execute('ROLLBACK')
            logger.error(f"Error clearing database: {e}")
            raise DispatcherException(f"Error clearing database: {e}")
```